list.h

```diff
@@ -0,0 +1,252 @@
+#ifndef _LINUX_LIST_H
+#define _LINUX_LIST_H
```

Comment:
@BC-SDS Thanks for reviewing. I was planning to use list implementation from ccan (https://github.com/rustyrussell/ccan/tree/master/ccan/list). Had linux kernel list already checked out, so used that. How about using from ccan? Its BSD-MIT license.